sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/OfferAccepter.java

```diff
@@ -35,10 +36,16 @@ public OfferAccepter addRecorder(OperationRecorder recorder) {
         return this;
     }
 
-    public List<OfferID> accept(SchedulerDriver driver, List<OfferRecommendation> recommendations) {
+    public List<OfferID> accept(List<OfferRecommendation> recommendations) {
         if (CollectionUtils.isEmpty(recommendations)) {
             LOGGER.warn("No recommendations, nothing to do");
-            return new ArrayList<>();
+            return Collections.emptyList();
+        }
+
+        Optional<SchedulerDriver> driver = Driver.getDriver();
+        if (!driver.isPresent()) {
+            LOGGER.error("No driver present for accepting offers.  This should never happen.");
```

Comment:
Is logging an `error` sufficient here? We could also print a stack trace, for example -- if we don't throw and exception and exit.